apps/o2-blocks/src/github-issue-template-button/editor.js

```diff
@@ -0,0 +1,154 @@
+		/>
+	</svg>
+);
+
+const Button = ( props ) => (
+	<WordpressButton
+		isPrimary
+		style={ { display: 'flex', justifyContent: 'center', alignItems: 'center' } }
+		href={ props.url }
+	>
+		{ props.children }
+		{ props.label }
+	</WordpressButton>
+);
+
+const GithubButton = ( props ) => (
+	<Button { ...props }>
+		<Icon icon={ icon } style={ { marginRight: '5px' } } />
+	</Button>
+);
+
+registerBlockType( 'a8c/github-issue-template-button', {
+	title: __( 'Github Isssue Template Button', 'a8c' ),
+	icon,
+	category: 'layout',
+	attributes: {
+		userOrOrg: {
+			type: 'string',
+			default: 'automattic',
+		},
+		repo: {
+			type: 'string',
+		},
+		title: {
+			type: 'string',
+		},
+		body: {
+			type: 'string',
+		},
+		buttonLabel: {
+			type: 'string',
+			default: 'Create Issue',
+		},
+		align: {
+			type: 'string',
+			default: 'left',
+		},
+	},
+	supports: {
+		align: [ 'left', 'center', 'right' ],
+	},
+	edit: ( props ) => {
+		const {
+			attributes: { userOrOrg, repo, title, body, buttonLabel, align },
+		} = props;
+
+		const requiredFieldsFilled = userOrOrg && repo;
+
+		const UnconfiguredButton = () => <Button label="⚙ Configure me first" />;
+
+		const onChangeUserOrOrg = ( newUserOrOrg ) => {
+			props.setAttributes( { userOrOrg: newUserOrOrg } );
+		};
+
+		const onChangeRepoName = ( newRepo ) => {
+			props.setAttributes( { repo: newRepo } );
+		};
+
+		const onChangeTitle = ( newTitle ) => {
+			props.setAttributes( { title: newTitle } );
+		};
+
+		const onChangeBody = ( newBody ) => {
+			props.setAttributes( { body: newBody } );
+		};
+
+		const onChangeButtonLabel = ( newButtonLabel ) => {
+			props.setAttributes( { buttonLabel: newButtonLabel } );
+		};
+
+		return (
+			<>
+				{ requiredFieldsFilled ? <GithubButton label={ buttonLabel } /> : <UnconfiguredButton /> }
+				<InspectorControls>
+					<TextControl
+						label={ __( 'User or Organization*' ) }
+						help="*This field is required"
+						onChange={ onChangeUserOrOrg }
+						value={ userOrOrg }
+					/>
+					<TextControl
+						label={ __( 'Repository Name*' ) }
+						help="*This field is required"
+						onChange={ onChangeRepoName }
+						value={ repo }
+					/>
+					<TextControl label={ __( 'Issue Title' ) } onChange={ onChangeTitle } value={ title } />
+					<TextareaControl label={ __( 'Issue Body' ) } onChange={ onChangeBody } value={ body } />
+					<TextControl
+						label={ __( 'Button Label' ) }
+						onChange={ onChangeButtonLabel }
+						value={ buttonLabel }
+					/>
+				</InspectorControls>
+			</>
+		);
+	},
+	save: ( props ) => {
+		const {
+			attributes: { userOrOrg, repo, title, body, buttonLabel, align },
+		} = props;
+
+		const requiredFieldsFilled = userOrOrg && repo;
+		if ( requiredFieldsFilled ) {
+			const url = newGithubIssueUrl( {
+				title,
+				repo,
+				user: userOrOrg,
+				body,
+			} );
+
+			return (
+				<div style={ { display: 'flex', justifyContent: align, alignItems: 'center' } }>
```

Comment:
Can we use className instead of inline styles here?

Reply:
The problem is that the class would end up not being available in the frontend. The README says to use a 'view.scss' for frontend styles but doesn't mention how to include it. I've left a comment here: p1595379050090900-slack-good-mountain. None of the other O2 blocks use a `view.scss` and or `view.js`. I'll ask around to get to know more about how this works and if needed, add the assets in the plugin.